### DISCUSSION TITLE
How to Debug grblHAL with VSCode on Pi Pico

### DISCUSSION THREAD

**Discussion by bear-fighter** [grblHAL/RP2040#51] [Q&A]
I'm trying to debug grblHAL to view how some variable get set using a pi pico.  The debug settings don't seem to be working because when I set breakpoints, they are greyed out and it says "no source file named /..../main.c"

**Comment by terjeio** [comment]
I do not know how to set up VSCode - it worked "out-of-the-box" for me on the Raspberry Pi 400 I use for development.
Perhaps someone with better VSCode knowledge can help, someone on the[ Raspberry Pi forum](https://forums.raspberrypi.com/viewforum.php?f=143) or even a [general search](https://duckduckgo.com/?t=ffab&q=vscode+debug+main.c+not+found&ia=web)?
